pipeline/vars/bm_node_reimage.bash

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# This script will use Teuthology framework to reimage
+# given set of nodes to the os version required
+# This script accepts the hostnames of the nodes as arg
+
+# To invoke this script, pass the argument as below
+# => bm_node_reimage.bash node1 node2 node3 os-version:9.0
+# last parameter 'os-version:<version>' defines the version to
+# which the nodes has to be reimaged to
+
+unlock_and_lock_node()
+{
+  node=$1
+  curr_user=$2
+  echo "The node $node is currenlty locked by $curr_user"
+  echo "Performing unlock"
+  cmd="source $v_env_path/bin/activate;teuthology-lock --unlock --owner $curr_user $node;deactivate"
+  res=$(eval $cmd)
+
+  # Now that the node is unlocked, perform lock
+  root_user="jenkins@magna006"
+  cmd="source $v_env_path/bin/activate;teuthology-lock --lock --owner root_user $node;deactivate"
+  res=$(eval $cmd)
+
+  # Verify whether the lock operation succeeded
+  cmd="source $v_env_path/bin/activate;teuthology-lock --list $node |
+  sed -nr 's/.*locked_by\": \"(.*)\",.*/\1/p';deactivate"
+  curr_user=$(eval $cmd)
+  if ["$curr_user" != "$root_user" ]; then
+    echo "Failed to unlock and lock the node. Exiting!!"
+    exit 1
+  else
+    echo "Successfully relocked machine under $root_user"
+  fi
+}
+
+
+v_env_path="/root/teuthology/virtualenv"
+
+# For each node, to perform Teuthology reimage, it has
+# to be locked under the current user's name
+for arg in "$@"
+do
+    if [[ "$arg" == *"os-version"* ]]; then
+      os_param=$arg
+    else
+      # Get current user
+      host=$arg
+      cmd="source $v_env_path/bin/activate;teuthology-lock --list $host | sed -nr 's/.*locked_by\": \"(.*)\",.*/\1/p';deactivate"
```

Comment:
We could do `~/.venv/bin/teuthology-lock --list $host | sed -nr 's/.*locked_by\":...`

IMO, by default the systems must be locked `jenkins-build@magna006`. We can provide support for overriding this user... other than that I believe it would go against the policy of lock.